core/src/Kokkos_BitManipulation.hpp

```diff
@@ -47,7 +47,7 @@ KOKKOS_FUNCTION constexpr T byteswap_fallback(T x) {
       lo_mask <<= CHAR_BIT;
       hi_mask >>= CHAR_BIT;
 
-      shift -= 2 * CHAR_BIT;
+      shift -= size_t{2} * CHAR_BIT;
```

Comment:
It seems that elsewhere you prefered to `static_cast` ?

Reply:
I really wanted to write `2z` but this only comes with C++23.
I can change it to a static_cast that's fine.